consensus/state.go

```diff
@@ -1295,6 +1295,57 @@ func (cs *State) defaultDoPrevote(height int64, round int32) {
 		return
 	}
 
+	/// The following implements line 28 - 33 of algorithm:
+	//
+	// 	upon ⟨PROPOSAL, h_p, round_p, v, vr⟩ from proposer(h_p, round_p)
+	// 	AND 2f + 1 ⟨PREVOTE, h_p, vr, id(v)⟩
+	// 	while step_p = propose ∧ (vr ≥ 0 ∧ vr < round_p) do {
+	// 	  if valid(v) ∧ (lockedRound_p ≤ vr ∨ lockedValue_p = v) {
+	// 		  broadcast ⟨PREVOTE, h_p, round_p, id(v)⟩
+	// 	  } else {
+	// 		  broadcast ⟨PREVOTE, h_p, round_p, nil⟩
+	// 	  }
+	// 	  step_p ← prevote
+	//  }
+	//
+	// Determine if the proposed block has a sane, non-nil proof-of-lock.
+	if cs.Proposal.POLRound >= 0 && cs.Proposal.POLRound < cs.Round {
+		// Validate the proof-of-lock using known prevotes.
+		blockID, ok := cs.Votes.Prevotes(cs.Proposal.POLRound).TwoThirdsMajority()
+		if ok && cs.ProposalBlock.HashesTo(blockID.Hash) {
+			// Validate the proof-of-lock round is at least as new as the possible locked round.
+			// (vr >= 0, vr > round_p, 2f+1 prevotes at round vr, lockedRound_p <= vr) execute 30.
+			if cs.Proposal.POLRound >= cs.LockedRound {
```

Comment:
Before we merge the [locked block fix](https://linear.app/dydx/issue/CORE-434/merge-locked-block-logic-fix), we will only reach here when `LockedRound == -1` and `LockedBlock == nil` . 

In the context of the recent liveness outage, this means that at round 1:
- validators who precommited and locked during round 0 will continue to vote for `B0` (duet to  line 1275) 
- validators who are not locked will also prevote for `B0` since they will verify at `POLRound == 0`, they also witnessed 2/3+ prevotes